src/main/java/baseball/domain/NumberComparator.java

```diff
@@ -0,0 +1,50 @@
+package baseball.domain;
+
+import java.util.List;
+
+import baseball.constant.GameConstants;
+
+public class NumberComparator {
+	private final List<Integer> answer;
+	private boolean isCorrect = false;
+	private int ballsCount;
+	private int strikesCount;
+
+	public NumberComparator(List<Integer> answer) {
+		this.answer = answer;
+	}
+
+	private int calculateStrikesCount(List<Integer> number) {
+		return (int)number.stream()
+			.filter(digit -> digit.equals(answer.get(number.indexOf(digit))))
+			.count();
+	}
+
+	private int calculateBallsCount(List<Integer> number) {
+		return (int)number.stream()
+			.filter(digit -> !digit.equals(answer.get(number.indexOf(digit))))
+			.filter(answer::contains)
+			.count();
+	}
+
+	public void compare(List<Integer> digits) {
```

Comment:
```suggestion
	public boolean compare(List<Integer> digits) {
```

isCorrect = true; 대신 isCorrect 결과를 반환하게 만들면
isCorrect 인스턴스 변수를 사용안하고 구현할 수 있을것 같습니다.